# HG changeset patch
# User Julien Bect <julien.bect@supelec.fr>
# Date 1361459998 -3600
#      Thu Feb 21 16:19:58 2013 +0100
# Node ID f89760972bf493a334d7afeed33da8ec6adb0997
# Parent  979ebfdd240d6e66d7ee5ca7c35011a37ff3f4f2
plotmatrix.m: Fix bug when number of columns in 2 inputs differs (bug #38368)

* scripts/plot/plotmatrix.m (__plotmatrix__): Reverse loop bounds of inner
and outer for loops.

diff --git a/scripts/plot/plotmatrix.m b/scripts/plot/plotmatrix.m
--- a/scripts/plot/plotmatrix.m
+++ b/scripts/plot/plotmatrix.m
@@ -159,19 +159,19 @@ function [h, ax, p, pax, need_usage] = _
 
   xsize = 0.9 / m;
   ysize = 0.9 / n;
   xoff = 0.05;
   yoff = 0.05;
   border = [0.130, 0.110, 0.225, 0.185] .* [xsize, ysize, xsize, ysize];
   border(3:4) = - border(3:4) - border(1:2);
 
-  for i = 1 : n
-    for j = 1 : m
-      pos = [xsize * (j - 1) + xoff, ysize * (n - i) + yoff, xsize, ysize];
+  for i = 1 : m
+    for j = 1 : n
+      pos = [xsize * (i - 1) + xoff, ysize * (n - j) + yoff, xsize, ysize];
       tmp = axes ("outerposition", pos, "position", pos + border,
                   "parent", parent);
       if (i == j && have_hist)
         pax = [pax ; tmp];
         [nn, xx] = hist (X(:, i));
         tmp = bar (xx, nn, 1.0);
         p = [p; tmp];
       else
